public_html/wp-content/themes/pattern-directory/src/hooks/route.js

```diff
@@ -58,6 +58,18 @@ export function RouteProvider( { children } ) {
 		setPath( newPath );
 	};
 
+	/**
+	 * Updates the path to the current browser path.
+	 */
+	const setPathOnPop = () => {
+		setPath( document.location.href );
+	};
+
+	useEffect( () => {
+		// When the browser modifies the history, update our path.
+		window.addEventListener( 'popstate', setPathOnPop ); // eslint-disable-line @wordpress/no-global-event-listener
```

Comment:
Should this also [unsubscribe the event listener](https://reactjs.org/docs/hooks-effect.html#example-using-hooks-1)?

Reply:
That's a good idea. 90076fe